src/java.base/share/classes/java/lang/classfile/CodeBuilder.java

```diff
@@ -550,66 +550,58 @@ default CodeBuilder arrayStore(TypeKind tk) {
-                        case CharType -> i2c();
-                        case ShortType -> i2s();
-                        case BooleanType -> iconst_1().iand();
-                        case VoidType, ReferenceType ->
-                            throw new IllegalArgumentException(String.format("convert %s -> %s", fromType, toType));
-                    };
-            case LongType ->
-                    switch (toType) {
-                        case IntType -> l2i();
-                        case LongType -> this;
-                        case DoubleType -> l2d();
-                        case FloatType -> l2f();
-                        case ByteType -> l2i().i2b();
-                        case CharType -> l2i().i2c();
-                        case ShortType -> l2i().i2s();
-                        case BooleanType -> l2i().iconst_1().iand();
-                        case VoidType, ReferenceType ->
-                            throw new IllegalArgumentException(String.format("convert %s -> %s", fromType, toType));
-                    };
-            case DoubleType ->
-                    switch (toType) {
-                        case IntType -> d2i();
-                        case LongType -> d2l();
-                        case DoubleType -> this;
-                        case FloatType -> d2f();
-                        case ByteType -> d2i().i2b();
-                        case CharType -> d2i().i2c();
-                        case ShortType -> d2i().i2s();
-                        case BooleanType -> d2i().iconst_1().iand();
-                        case VoidType, ReferenceType ->
-                            throw new IllegalArgumentException(String.format("convert %s -> %s", fromType, toType));
-                    };
-            case FloatType ->
-                    switch (toType) {
-                        case IntType -> f2i();
-                        case LongType -> f2l();
-                        case DoubleType -> f2d();
-                        case FloatType -> this;
-                        case ByteType -> f2i().i2b();
-                        case CharType -> f2i().i2c();
-                        case ShortType -> f2i().i2s();
-                        case BooleanType -> f2i().iconst_1().iand();
-                        case VoidType, ReferenceType ->
-                            throw new IllegalArgumentException(String.format("convert %s -> %s", fromType, toType));
-                    };
-            case VoidType, ReferenceType ->
-                throw new IllegalArgumentException(String.format("convert %s -> %s", fromType, toType));
-        };
+        var computationalFrom = fromType.asLoadable();
+        var computationalTo = toType.asLoadable();
+        if (computationalFrom != computationalTo) {
+            switch (computationalTo) {
+                case INT -> {
+                    switch (computationalFrom) {
+                        case FLOAT -> f2i();
```

Comment:
Actual implementation shows complete conversion table as nested switches.
What is the advantage of the proposed change?

Reply:
Smaller switches reduce code size. A single switch case in the range adds 4 bytes to the size. This may be further optimized later.